hypothesis-python/tests/conjecture/common.py

```diff
@@ -140,6 +140,27 @@ def clamped_shrink_towards(kwargs):
     return v
 
 
+@st.composite
+def integer_weights(draw, min_value=None, max_value=None):
+    # Sampler doesn't play well with super small floats, so exclude them
+    weights = draw(
+        st.dictionaries(
+            st.integers(min_value=min_value, max_value=max_value),
+            st.floats(0.001, 1),
+            max_size=255,
+        )
+    )
+    # invalid to have a weighting that disallows all possibilities
+    assume(sum(weights.values()) != 0)
+    # re-normalize probabilities to sum to some arbitrary target < 1
+    target = draw(st.floats(0.001, 0.999))
+    factor = target / sum(weights.values())
+    weights = {k: v * factor for k, v in weights.items()}
+    # float rounding error can cause this to fail.
+    assume(sum(weights.values()) == target)
```

Comment:
Given that it's arbitrary anyway, surely we don't really care about this condition?  I'd drop it for performance.

(OK, and just noticed I'm commenting on moved rather than new code.  Nonblocking then)

Reply:
I think this used to target 1 at some point - good comments thanks